converter.go

```diff
@@ -388,6 +387,345 @@
+	}
+	return bindingValue{&jsonString, "json", &schema}, nil
+}
+
+func toNullableInt64(val any) (int64, bool) {
+	switch v := val.(type) {
+	case sql.NullByte:
+		return int64(v.Byte), v.Valid
+	case sql.NullInt16:
+		return int64(v.Int16), v.Valid
+	case sql.NullInt32:
+		return int64(v.Int32), v.Valid
+	case sql.NullInt64:
+		return v.Int64, v.Valid
+	}
+	// should never happen, the list above is exhaustive
+	panic("Only byte, int16, int32 or int64 are supported")
+}
+
+func toNullableTime(val any) (time.Time, bool, error) {
+	switch v := val.(type) {
+	case time.Time:
+		return v, true, nil
+	case sql.NullTime:
+		return v.Time, v.Valid, nil
+	}
+	return time.Now(), false, fmt.Errorf("cannot use %T as time", val)
+}
+
+func stringOrIntToString(v reflect.Value) string {
+	if v.CanInt() {
+		return strconv.Itoa(int(v.Int()))
+	}
+	return v.String()
+}
+
+func goTypeToFieldMetadata(typ reflect.Type, tsmode snowflakeType, params map[string]*string) (fieldMetadata, error) {
+	if tsmode == binaryType {
+		return fieldMetadata{
+			Type:     "BINARY",
+			Nullable: true,
+		}, nil
+	}
+	if typ.Kind() == reflect.Pointer {
+		typ = typ.Elem()
+	}
+	switch typ.Kind() {
+	case reflect.String:
+		return fieldMetadata{
+			Type:     "TEXT",
+			Nullable: true,
+		}, nil
+	case reflect.Bool:
+		return fieldMetadata{
+			Type:     "BOOLEAN",
+			Nullable: true,
+		}, nil
+	case reflect.Int, reflect.Int8, reflect.Uint8, reflect.Int16, reflect.Int32, reflect.Int64:
+		return fieldMetadata{
+			Type:      "FIXED",
+			Precision: 38,
+			Nullable:  true,
+		}, nil
+	case reflect.Float32, reflect.Float64:
+		return fieldMetadata{
+			Type:     "REAL",
+			Nullable: true,
+		}, nil
+	case reflect.Struct:
+		if typ.AssignableTo(reflect.TypeOf(sql.NullString{})) {
+			return fieldMetadata{
+				Type:     "TEXT",
+				Nullable: true,
+			}, nil
+		} else if typ.AssignableTo(reflect.TypeOf(sql.NullBool{})) {
+			return fieldMetadata{
+				Type:     "BOOLEAN",
+				Nullable: true,
+			}, nil
+		} else if typ.AssignableTo(reflect.TypeOf(sql.NullByte{})) || typ.AssignableTo(reflect.TypeOf(sql.NullInt16{})) || typ.AssignableTo(reflect.TypeOf(sql.NullInt32{})) || typ.AssignableTo(reflect.TypeOf(sql.NullInt64{})) {
+			return fieldMetadata{
+				Type:      "FIXED",
+				Precision: 38,
+				Nullable:  true,
+			}, nil
+		} else if typ.AssignableTo(reflect.TypeOf(sql.NullFloat64{})) {
+			return fieldMetadata{
+				Type:     "REAL",
+				Nullable: true,
+			}, nil
+		} else if tsmode == dateType {
+			return fieldMetadata{
+				Type:     "DATE",
+				Nullable: true,
+			}, nil
+		} else if tsmode == timeType {
+			return fieldMetadata{
+				Type:     "TIME",
+				Nullable: true,
+			}, nil
+		} else if tsmode == timestampTzType {
+			return fieldMetadata{
+				Type:     "TIMESTAMP_TZ",
+				Nullable: true,
+			}, nil
+		} else if tsmode == timestampNtzType {
+			return fieldMetadata{
+				Type:     "TIMESTAMP_NTZ",
+				Scale:    9,
```

Comment:
Is this a default? Why 9?

Reply:
It seems that this is not required at all.